documentation/SLACK/SLACK_TROUBLESHOOTING.md

```diff
@@ -0,0 +1,883 @@
+     --description "Testing bidirectional sync" \
+     --severity "Medium"
+   ```
+
+2. **Verify Channel Creation**:
+   - Check Slack for new channel: `aws-security-incident-response-case-<caseId>`
+   - Verify initial notification posted
+
+3. **Check Logs**:
+   ```bash
+   aws logs tail /aws/lambda/SecurityIncidentResponseSlackClient --follow
+   ```
+
+4. **Verify DynamoDB**:
+   ```bash
+   aws dynamodb scan \
+     --table-name <IncidentsTableName> \
+     --filter-expression "attribute_exists(slackChannelId)"
+   ```
+
+#### Test 2: Slack to AWS SIR Flow
+
+1. **Post Message in Channel**:
+   - Go to incident channel in Slack
+   - Post a test message
+
+2. **Verify Comment in AWS SIR**:
+   ```bash
+   aws security-ir list-comments --case-id <caseId>
+   ```
+
+3. **Check Logs**:
+   ```bash
+   aws logs tail /aws/lambda/SlackEventsBoltHandler --follow
+   ```
+
+#### Test 3: Slash Commands
+
+1. **Test Status Command**:
+   - In incident channel, type: `/security-ir status`
+   - Verify response with case details
+
+2. **Test Update Command**:
+   - Type: `/security-ir update-status Investigating`
+   - Verify case status updated in AWS SIR
+
+3. **Check Logs**:
+   ```bash
+   aws logs tail /aws/lambda/SlackCommandHandler --follow
+   ```
+
+### Performance Testing
+
+#### Monitor Lambda Performance
+
+1. **Check Execution Duration**:
+   ```bash
+   aws cloudwatch get-metric-statistics \
+     --namespace AWS/Lambda \
+     --metric-name Duration \
+     --dimensions Name=FunctionName,Value=SecurityIncidentResponseSlackClient \
+     --start-time $(date -u -d '1 hour ago' +%Y-%m-%dT%H:%M:%S) \
+     --end-time $(date -u +%Y-%m-%dT%H:%M:%S) \
+     --period 300 \
+     --statistics Average,Maximum
+   ```
+
+2. **Check Error Rate**:
+   ```bash
+   aws cloudwatch get-metric-statistics \
+     --namespace AWS/Lambda \
+     --metric-name Errors \
+     --dimensions Name=FunctionName,Value=SecurityIncidentResponseSlackClient \
+     --start-time $(date -u -d '1 hour ago' +%Y-%m-%dT%H:%M:%S) \
+     --end-time $(date -u +%Y-%m-%dT%H:%M:%S) \
+     --period 300 \
+     --statistics Sum
+   ```
+
+#### Monitor API Gateway Performance
+
+1. **Check API Latency**:
+   ```bash
+   aws cloudwatch get-metric-statistics \
+     --namespace AWS/ApiGateway \
+     --metric-name Latency \
+     --dimensions Name=ApiName,Value=SlackEventsApi \
+     --start-time $(date -u -d '1 hour ago' +%Y-%m-%dT%H:%M:%S) \
+     --end-time $(date -u +%Y-%m-%dT%H:%M:%S) \
+     --period 300 \
+     --statistics Average,Maximum
+   ```
+
+2. **Check API Errors**:
+   ```bash
+   aws cloudwatch get-metric-statistics \
+     --namespace AWS/ApiGateway \
+     --metric-name 5XXError \
+     --dimensions Name=ApiName,Value=SlackEventsApi \
+     --start-time $(date -u -d '1 hour ago' +%Y-%m-%dT%H:%M:%S) \
+     --end-time $(date -u +%Y-%m-%dT%H:%M:%S) \
+     --period 300 \
+     --statistics Sum
+   ```
+
+## Health Checks
+
+### Daily Health Check Checklist
```

Comment:
issue(blocking): This reads like we expect the customer to perform these checks every day.  Let's create alarms for this and just tell the customer when something is broken.